--- hylafax.orig/faxd/faxQueueApp.c++	Mon Jun 18 00:53:56 2001
+++ hylafax/faxd/faxQueueApp.c++	Tue Jun 19 19:45:10 2001
@@ -1406,14 +1406,21 @@
 		: (requeueInterval>>1) + (random()%requeueInterval));
 	}
 	/*
-	 * Bump the job priority so the retry will get processed
-	 * before new jobs.  We bound the priority to keep it within
-	 * a fixed ``range'' around it's starting priority.  This
-	 * is intended to keep jobs in different ``classifications''
-	 * from conflicting (e.g. raising a bulk-style fax up into
-	 * the priority range of a non-bulk-style fax).
+	 * Bump the job priority if is not bulk-style in which case
+	 * we dip the job job priority.  This is intended to prevent
+	 * non-bulk faxes from becoming buried by new jobs which
+	 * could prevent a timely retry.  However, it is also intended
+	 * to allow all bulk faxes to be attempted before retrying
+	 * any that could not complete on the first attempt.  This 
+	 * aids in timely delivery of bulk faxes as a group rather than
+	 * preoccupation with individual jobs as is the case with 
+	 * non-bulk style jobs.  We bound the priority to keep it
+	 * within a fixed "range" around it's starting priority.  This
+	 * is intended to keep "normal" and "high" priority jobs
+	 * from conflicting.
 	 */
-	if (JOBHASH(job.pri-1) == JOBHASH(req->usrpri))
+	if (job.pri != 255 && job.pri > 190) job.pri++;
+	else if (JOBHASH(job.pri-1) == JOBHASH(req->usrpri))
 	    job.pri--; 
 	job.state = (req->tts > now) ?
 	    FaxRequest::state_sleeping : FaxRequest::state_ready;
--- hylafax.orig/man/sendfax.1	Mon Jun 18 00:53:57 2001
+++ hylafax/man/sendfax.1	Tue Jun 19 19:55:42 2001
@@ -444,9 +444,14 @@
 A priority is an integer value in the range [0-255].
 Lower values result in higher priority processing.
 By default, each job is assigned 127 for an initial scheduling priority.
-A job's priority is raised (numerically lowered) for each
+If a job's priority is greater than bulk faxes then the
+job's priority is raised (numerically lowered) for each
 failed attempt to transmit so that retransmit attempts
 are done ahead of newly submitted jobs.
+In the case of bulk faxes (numerically higher than 190), then the
+job's priority is lowered (numerically raised) for each failed
+attempt to transmit so that the whole of the bulk-queued jobs
+will be attempted before any are retried.
 The
 .I priority
 may also be specified as one of the following symbolic names:
@@ -457,7 +462,7 @@
 .I bulk
 or
 .I junk
-(207),
+(191),
 .I high
 (63).
 .TP 12
--- hylafax.orig/man/hylafax-server.4f	Mon Jun 18 00:53:57 2001
+++ hylafax/man/hylafax-server.4f	Tue Jun 19 19:59:09 2001
@@ -548,10 +548,10 @@
 being processed using the
 .IR faxalter (${MANNUM1_8})
 program.
-A job's scheduling priority may also be raised by
+A job's scheduling priority may also be altered by
 .I faxq
 in response to certain scheduling events (e.g. after a failed
-attempt to send the priority may be raised).
+attempt to send).
 .PP
 Modems are assigned to outbound jobs if they are deemed ready
 for use.
