--- hylafax-6.0.6/faxd/Makefile.in.orig	2018-04-14 21:55:33.933554074 +0200
+++ hylafax-6.0.6/faxd/Makefile.in	2018-04-14 21:59:53.600217776 +0200
@@ -172,6 +172,9 @@
 
 LIBFAXSERVER=-L. -lfaxserver-${ABI_VERSION}
 
+${SERVER_OBJS} ${MODEM_OBJS}: %.o: %.c++
+	${C++F} -fPIC ${C++FILE} -c $<
+
 libfaxserver-${ABI_VERSION}.a: ${SERVER_OBJS} ${MODEM_OBJS}
 	${AR} ${AROPTS} libfaxserver-${ABI_VERSION}.a $?
 	${RANLIB} libfaxserver-${ABI_VERSION}.a
--- hylafax-4.1.8/regex/Makefile.in.orig	1999-08-05 10:01:44.000000000 +0000
+++ hylafax-4.1.8/regex/Makefile.in	2004-03-24 12:07:30.000000000 +0000
@@ -53,6 +53,9 @@
 
 @MAKEINCLUDE@ @MAKELQUOTE@${COMMONRULES}@MAKERQUOTE@
 
+${OBJECTS}: %.o: %.c
+	${CCF} -fPIC -c $<
+
 libregex.a: ${OBJECTS}
 	@rm -f $@;
 	${AR} ${AROPTS} $@ ${OBJECTS}
