--- hylafax-4.4.5/util/b64-encode.awk~	2009-03-20 16:36:34.000000000 +0100
+++ hylafax-4.4.5/util/b64-encode.awk	2010-08-16 08:20:05.990925413 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/awk -f
+#!/bin/awk -f
 #	$Id$
 #
 # HylaFAX Facsimile Software
--- /home/users/arekm/rpm/BUILD/hylafax-4.4.5/util/qp-encode.awk~	2007-01-09 15:14:12.000000000 +0100
+++ /home/users/arekm/rpm/BUILD/hylafax-4.4.5/util/qp-encode.awk	2010-08-16 08:20:31.653423097 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/awk -f
+#!/bin/awk -f
 #	$Id$
 #
 # HylaFAX Facsimile Software
