--- hylafax-4.1.7.org/hfaxd/FileTransfer.c++	2003-02-08 23:49:52.000000000 +0100
+++ hylafax-4.1.7/hfaxd/FileTransfer.c++	2003-10-08 15:04:10.000000000 +0200
@@ -556,7 +556,7 @@
 	} else
 	    perror_reply(553, filename, errno);
     } else
-	reply(553, emsg);
+	reply(553, "%s", (const char*)emsg);
 }
 
 /*
--- hylafax-4.1.7.org/hfaxd/Parser.c++	2003-02-08 23:49:53.000000000 +0100
+++ hylafax-4.1.7/hfaxd/Parser.c++	2003-10-08 15:05:16.000000000 +0200
@@ -963,7 +963,7 @@
 	break;
     case T_TRIGGER:
 	if (string_param(s, "trigger-spec")) {
-	    logcmd(t, s);
+	    logcmd(t, "%s", (const char*) s);
 	    triggerCmd("%s", (const char*) s);
 	    return (true);
 	}
--- hylafax-4.1.7.org/hfaxd/SNPPServer.c++	2003-02-08 23:49:53.000000000 +0100
+++ hylafax-4.1.7/hfaxd/SNPPServer.c++	2003-10-08 15:06:38.000000000 +0200
@@ -849,7 +849,7 @@
 	fxStr emsg;
 	u_int seqnum = getDocumentNumbers(1, emsg);
 	if (seqnum == (u_int) -1) {
-	    reply(554, emsg);
+	    reply(554, "%s", (const char*)emsg);
 	    return;
 	}
 	msgFile = fxStr::format("/%s/doc%u.page", FAX_TMPDIR, seqnum);
@@ -1053,7 +1053,7 @@
 	fxStr emsg;
 	u_int seqnum = getDocumentNumbers(1, emsg);
 	if (seqnum == (u_int) -1) {
-	    reply(554, emsg);
+	    reply(554, "%s", (const char*)emsg);
 	    return;
 	}
 	msgFile = fxStr::format("/%s/doc%u.page", FAX_TMPDIR, seqnum);
