--- hylafax-6.0.6/libhylafax/FaxRecvInfo.c++.orig	2012-06-06 02:58:38.000000000 +0200
+++ hylafax-6.0.6/libhylafax/FaxRecvInfo.c++	2018-04-16 19:01:56.584228628 +0200
@@ -96,26 +96,27 @@
     cp = strchr(cp+1, '"');
     if (cp == NULL || cp[1] != ',' || cp[2] != '"')
 	return (false);
-    passwd = cp+1;
+    passwd = cp+3;
     passwd.resize(sender.next(0,'"'));
-    cp = strchr(cp+1, '"');
+    cp = strchr(cp+3, '"');
     if (cp == NULL || cp[1] != ',' || cp[2] != '"')
 	return (false);
     subaddr = cp+3;			// +1 for "/+1 for ,/+1 for "
     subaddr.resize(subaddr.next(0,'"'));
-    cp = strchr(cp+1, '"');
+    cp = strchr(cp+3, '"');
     if (cp == NULL || cp[1] != ',' || cp[2] != '"')
 	return (false);
     reason = cp+3;			// +1 for "/+1 for ,/+1 for "
     reason.resize(reason.next(0,'"'));
-    cp = strchr(cp+1, '"');
+    cp = strchr(cp+3, '"');
     if (cp == NULL || cp[1] != ',' || cp[2] != '"')
 	return (false);
     u_int i = 0;
-    while (cp+2 != '\0') {
+    while (cp[1] == ',' && cp[2] == '"') {
 	callid[i] = cp+3;		// +1 for "/+1 for ,/+1 for "
-	if (*cp == '\"') break;
 	callid[i].resize(callid[i].next(0,'"'));
+	cp = strchr(cp+3, '"');
+	if (cp == NULL) break;
 	i++;
     }
     return (true);
