diff -Naur hylafax-4.2.1.orig/faxd/MemoryDecoder.c++ hylafax-4.2.1/faxd/MemoryDecoder.c++
--- hylafax-4.2.1.orig/faxd/MemoryDecoder.c++	2005-01-11 20:45:08.133122704 +0100
+++ hylafax-4.2.1/faxd/MemoryDecoder.c++	2005-01-11 20:45:11.835559848 +0100
@@ -147,6 +147,11 @@
 		topMargin *= 4;
 		break;
 	}
+        /*
+         * Skip at least 200 rows. ZYXELs like it.
+        */
+        if( topMargin < 220 ) topMargin = 220;
+		
 	do {
 	    (void) decodeRow(NULL, rowpixels);
 	} while (--topMargin);
