diff -uNr hylafax-4.1beta3/faxd/Makefile.LINUXdso hylafax-4.1beta3_new/faxd/Makefile.LINUXdso
--- hylafax-4.1beta3/faxd/Makefile.LINUXdso	Thu Mar  8 23:35:53 2001
+++ hylafax-4.1beta3_new/faxd/Makefile.LINUXdso	Thu Mar  8 23:33:59 2001
@@ -33,14 +33,14 @@
 
 LIBVERSION=4.0.1
 
-libfaxserver.${DSO}: ${SERVER_OBJS} ${MODEM_OBJS}
+libfaxserver.${DSO}.${LIBVERSION}: ${SERVER_OBJS} ${MODEM_OBJS}
 	${C++} ${DSOOPTS} -o libfaxserver.${DSO}.${LIBVERSION} \
 	    ${SERVER_OBJS} ${MODEM_OBJS} ${LLDLIBS}
-	ln -s libfaxserver.${DSO}.${LIBVERSION} libfaxserver.${DSO}
 
-installDSO: libfaxserver.${DSO}
+	ln -sf libfaxserver.${DSO}.${LIBVERSION} libfaxserver.${DSO}
+
+installDSO: libfaxserver.${DSO}.${LIBVERSION}
 	${PUTSERV} -F ${LIBEXEC} -m 755 -O libfaxserver.${DSO}.${LIBVERSION}
 	(cd ${LIBEXEC}; ln -s libfaxserver.${DSO}.${LIBVERSION} libfaxserver.${DSO})
-
 
 
diff -uNr hylafax-4.1beta3/faxd/Makefile.in hylafax-4.1beta3_new/faxd/Makefile.in
--- hylafax-4.1beta3/faxd/Makefile.in	Thu Mar  8 23:35:53 2001
+++ hylafax-4.1beta3_new/faxd/Makefile.in	Thu Mar  8 21:36:26 2001
@@ -42,6 +42,8 @@
 SRCDIR	= ${DEPTH}/${TOPSRCDIR}/faxd
 VPATH	= ../@SRCDIR@/faxd
 
+LIBVERSION=4.0.1 
+
 C++FILES=ClassModem.c++ \
 	FaxModem.c++ \
 	Class0.c++ \
@@ -143,7 +145,7 @@
 FAXSENDOBJS=  faxSendApp.o
 FAXQCLEANOBJS=faxQCleanApp.o
 FAXGETTYOBJS= Getty.o Getty@GETTY@.o faxGettyApp.o
-TARGETS=libfaxserver.${DSO} \
+TARGETS=libfaxserver.${DSO}.${LIBVERSION} \
 	faxq faxsend faxgetty pagesend faxqclean \
 	tsitest tagtest cqtest choptest
 
@@ -171,29 +173,29 @@
 faxQueueApp.o FaxRecv.o:
 incdepend: hash.h
 
-faxq:	${FAXQOBJS} libfaxserver.${DSO} ${LIBS}
+faxq:	${FAXQOBJS} libfaxserver.${DSO}.${LIBVERSION} ${LIBS}
 	${C++F} -o $@ ${FAXQOBJS} -lfaxserver ${LDFLAGS}
-faxsend: ${FAXSENDOBJS} libfaxserver.${DSO} ${LIBS}
+faxsend: ${FAXSENDOBJS} libfaxserver.${DSO}.${LIBVERSION} ${LIBS}
 	${C++F} -o $@ ${FAXSENDOBJS} -lfaxserver ${LDFLAGS}
-faxgetty: ${FAXGETTYOBJS} libfaxserver.${DSO} ${LIBS}
+faxgetty: ${FAXGETTYOBJS} libfaxserver.${DSO}.${LIBVERSION} ${LIBS}
 	${C++F} -o $@ ${FAXGETTYOBJS} -lfaxserver ${LDFLAGS}
-faxqclean: ${FAXQCLEANOBJS} libfaxserver.${DSO} ${LIBS}
+faxqclean: ${FAXQCLEANOBJS} libfaxserver.${DSO}.${LIBVERSION} ${LIBS}
 	${C++F} -o $@ ${FAXQCLEANOBJS} -lfaxserver ${LDFLAGS}
 
 PAGESENDOBJS=\
 	pageSendApp.o
-pagesend:${PAGESENDOBJS} libfaxserver.${DSO} ${LIBS}
+pagesend:${PAGESENDOBJS} libfaxserver.${DSO}.${LIBVERSION} ${LIBS}
 	${C++F} -o $@ ${PAGESENDOBJS} -lfaxserver ${LDFLAGS}
 
-tagtest: tagtest.o libfaxserver.${DSO} ${LIBS}
+tagtest: tagtest.o libfaxserver.${DSO}.${LIBVERSION} ${LIBS}
 	${C++F} -o $@ tagtest.o -lfaxserver ${LDFLAGS}
-cqtest:	cqtest.o libfaxserver.${DSO} ${LIBS}
+cqtest:	cqtest.o libfaxserver.${DSO}.${LIBVERSION} ${LIBS}
 	${C++F} -o $@ cqtest.o -lfaxserver ${LDFLAGS}
-choptest: choptest.o libfaxserver.${DSO} ${LIBS}
+choptest: choptest.o libfaxserver.${DSO}.${LIBVERSION} ${LIBS}
 	${C++F} -o $@ choptest.o -lfaxserver ${LDFLAGS}
-tsitest: tsitest.o libfaxserver.${DSO} ${LIBS}
+tsitest: tsitest.o libfaxserver.${DSO}.${LIBVERSION} ${LIBS}
 	${C++F} -o $@ tsitest.o -lfaxserver ${LDFLAGS}
-trigtest: trigtest.o libfaxserver.${DSO} ${LIBS}
+trigtest: trigtest.o libfaxserver.${DSO}.${LIBVERSION} ${LIBS}
 	${C++F} -o $@ trigtest.o -lfaxserver ${LDFLAGS}
 
 PUTSERV=${INSTALL} -idb ${PRODUCT}.sw.server
diff -uNr hylafax-4.1beta3/util/Makefile.LINUXdso hylafax-4.1beta3_new/util/Makefile.LINUXdso
--- hylafax-4.1beta3/util/Makefile.LINUXdso	Thu Mar  8 23:35:53 2001
+++ hylafax-4.1beta3_new/util/Makefile.LINUXdso	Thu Mar  8 23:34:08 2001
@@ -31,18 +31,18 @@
 #
 LIBVERSION=4.0.1
 
-libfaxutil.${DSO}: ${OBJECTS}
+libfaxutil.${DSO}.${LIBVERSION}: ${OBJECTS}
 	${C++} ${DSOOPTS} -o libfaxutil.${DSO}.${LIBVERSION} ${OBJECTS} \
 	    ${DSODELAY} ${LIBTIFF} ${DSODELAY} ${LIBZ} ${DSODELAY} ${LIBREGEX} \
 	    ${MACHDEPLIBS}
-	ln -s libfaxutil.${DSO}.${LIBVERSION} libfaxutil.${DSO}
 
-../util/libfaxutil.${DSO}: libfaxutil.${DSO}
+	ln -sf libfaxutil.${DSO}.${LIBVERSION} libfaxutil.${DSO}
 
-installDSO: libfaxutil.${DSO}
+../util/libfaxutil.${DSO}.${LIBVERSION}: libfaxutil.${DSO}.${LIBVERSION}
+
+installDSO: libfaxutil.${DSO}.${LIBVERSION}
 	${PUTCLIENT} -F ${LIBEXEC} -m 755 -O libfaxutil.${DSO}.${LIBVERSION}
 	(cd ${LIBEXEC}; ln -s libfaxutil.${DSO}.${LIBVERSION} libfaxutil.${DSO})
-
 
 
 
diff -uNr hylafax-4.1beta3/util/Makefile.in hylafax-4.1beta3_new/util/Makefile.in
--- hylafax-4.1beta3/util/Makefile.in	Sun Sep  3 05:39:34 2000
+++ hylafax-4.1beta3_new/util/Makefile.in	Thu Mar  8 21:36:26 2001
@@ -42,6 +42,8 @@
 SRCDIR	= ${DEPTH}/${TOPSRCDIR}/util
 VPATH	= ../@SRCDIR@/util
 
+LIBVERSION=4.0.1 
+
 C++FILES=Array.c++ \
 	BoolArray.c++ \
 	Dictionary.c++ \
@@ -86,7 +88,7 @@
 	SNPPClient.c++ \
 	SNPPJob.c++
 CFILES= cvtfacility.c fxassert.c
-TARGETS=libfaxutil.${DSO} faxmsg faxmodem faxadduser faxconfig faxdeluser \
+TARGETS=libfaxutil.${DSO}.${LIBVERSION} faxmsg faxmodem faxadduser faxconfig faxdeluser \
     faxstate faxinfo faxwatch textfmt dialtest typetest tiffcheck
 
 LC++INCS=${ZLIBINC}			# for FaxClient.c++
